core-bundle/src/Resources/contao/dca/tl_page.php

```diff
@@ -502,6 +502,7 @@
 			'filter'                  => true,
 			'inputType'               => 'checkbox',
 			'foreignKey'              => 'tl_member_group.name',
+			'options_callback'        => array('tl_page', 'getGroups'),
```

Comment:
New callbacks should always be implemented as a proper service and not added to the legacy `tl_` classes.

Reply:
Again, I really don't see any (or very little) sense in this rule. Having 40 callbacks in the DCA file and 2 in a listener just shatters the code and makes it less comprehensive. IMHO, the rule should be:

* If it is a new DCA file, add the callbacks as event listeners.
* If it is an existing DCA file, add the callbacks there. It most likely already contains dozens of callbacks, so one more does not make any difference.
* If you move callbacks of existing DCA files to event listeners, move them **all at once** and not just selectively.

Reply:
Nevertheless, changed in f9c8f3ee87897f7159f37dc68d96d8b1171c31e9.